C error when using or block after pushing to a channel

It seems that pushing to a channel is still treated as option (despite not in the document) but not implemented in the actual code. 

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Pro v19043 64-bit
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz  

CC version: Error: exec failed (CreateProcess) with code 2: 系统找不到指定的文件。

 cmd: cc --version

getwd: C:\Users\OliverCWY
vmodules: C:\Users\OliverCWY\.vmodules
vroot: C:\myPath\v
vexe: C:\myPath\v\v.exe
vexe mtime: 2022-08-16 19:54:22
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 6a728e1.8736f26

Git version: git version 2.34.1.windows.1
Git vroot status: weekly.2022.33-7-g8736f26a
.git/config present: true
thirdparty/tcc: N/A

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg D:\Downloads\test_channel.v`
```v
module main


fn main() {
    ch := chan int{cap: 100}
    ch.close()
    ch <- 1 or {
		println("Channel closed. ")
    }
}
```

**What did you expect to see?**

A transpile time error stating that or block should not be used here / "Channel closed. "


**What did you see instead?**
```
==================
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c: At top level:
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c:3312:15: error: unknown type name '_option_void'
 3312 | static inline _option_void __Option_chan_int_pushval(chan_int ch, int e) {
      |               ^~~~~~~~~~~~
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c: In function '__Option_chan_int_pushval':
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c:3314:25: error: '_option_void' undeclared (first use in this function); did you mean '_option_ok'?
 3314 |                 return (_option_void){ .state = 2, .err = _v_error(_SLIT("channel closed")), .data = {EMPTY_STRUCT_INITIALIZATION} };
      |                         ^~~~~~~~~~~~
      |                         _option_ok
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c:3314:25: note: each undeclared identifier is reported only once for each function it appears in
C:\Users\OLIVER~1\AppData\Local\Temp\v_0\test_channel.8546838419566819934.tmp.c:3314:38: error: expected ';' before '{' token
 3314 |                 return (_option_void){ .state = 2, .err = _v_error(_SLIT("channel closed")), .data = {EMPTY_STRUCT_INITIALIZATION} };
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -390,8 +390,6 @@ pub fn gen(files []&ast.File, table &ast.Table, pref &pref.Preferences) string {
 	}
 
 	global_g.gen_jsons()
-	global_g.write_optionals()
-	global_g.write_results()
 	global_g.dump_expr_definitions() // this uses global_g.get_str_fn, so it has to go before the below for loop
 	for i := 0; i < global_g.str_types.len; i++ {
 		global_g.final_gen_str(global_g.str_types[i])
@@ -406,6 +404,8 @@ pub fn gen(files []&ast.File, table &ast.Table, pref &pref.Preferences) string {
 	global_g.gen_array_index_methods()
 	global_g.gen_equality_fns()
 	global_g.gen_free_methods()
+	global_g.write_results()
+	global_g.write_optionals()
 	global_g.sort_globals_consts()
 	global_g.timers.show('cgen unification')
 

--- a/vlib/v/tests/inout/push_on_closed_channel.out
+++ b/vlib/v/tests/inout/push_on_closed_channel.out
@@ -1,0 +1,1 @@
+Channel closed.

--- a/vlib/v/tests/inout/push_on_closed_channel.vv
+++ b/vlib/v/tests/inout/push_on_closed_channel.vv
@@ -1,0 +1,7 @@
+module main
+
+fn main() {
+	ch := chan int{cap: 100}
+	ch.close()
+	ch <- 1 or { println('Channel closed.') }
+}

